src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+import type {ResourceViewsMap} from '../types';
+
+class ResourceViewRegistry {
+    resourceViews: ResourceViewsMap;
+
+    constructor() {
+        this.clear();
+    }
+
+    clear() {
+        this.resourceViews = {};
+    }
+
+    addResourceViews(resourceViews: ResourceViewsMap) {
+        Object.keys(resourceViews).forEach((resourceKey) => {
+            if (this.resourceViews[resourceKey]) {
```

Comment:
When we already have a `has` method, I would use it here

Reply:
The `has` is specific for the `view` of a resource key not if general the resourceKey exist.